confirm.c

```diff
@@ -763,16 +766,17 @@ confirm_query_create4(const struct sockaddr_storage *dst, uint8_t ttl,
 }
 
 struct confirm_query *
-confirm_query_create6(const struct sockaddr_storage *dst, uint8_t ttl,
-		uint8_t traffic_class, uint32_t flow_label,
-		uint16_t icmpid, uint8_t flowid,
-		confirm_query_cb cb)
+confirm_query_create6_icmp(const struct sockaddr_storage *dst, uint8_t ttl,
+		uint8_t traffic_class, uint32_t flow_label, uint16_t icmpid,
+		uint8_t flowid, confirm_query_cb cb)
 {
-	struct confirm_query *query;
-
-	query = malloc(sizeof(struct confirm_query));
+	struct confirm_query *query = malloc(sizeof(struct confirm_query));
```

Comment:
I know you got this from me, but it is preferable to have `sizeof(*query)`, because then it does not depend what the type of `query` is. Let's start preferring the new form on new code (and rewrite it whenever we come across one and it is easy to fix).